src/core/thread/mle.cpp

```diff
@@ -3073,6 +3076,16 @@ otError Mle::HandleLeaderData(const Message &aMessage, const Ip6::MessageInfo &a
         {
             SetLeaderData(leaderData.GetPartitionId(), leaderData.GetWeighting(), leaderData.GetLeaderRouterId());
             mRetrieveNewNetworkData = true;
+
+#if OPENTHREAD_FTD
+            if (IsFullThreadDevice())
+            {
+                routerIdSet.Clear();
+                routerIdSet.Add(GetLeaderId());
+                routerIdSet.Add(GetParent().GetRouterId());
+                Get<RouterTable>().UpdateRouterIdSet(0, routerIdSet);
```

Comment:
It is possible that Child can receive from parent a `Data Response` with different partition ID (in LeaderData) from the Child's current LeaderData. The original code only `SetLeaderData`, but didn't update the `RouterTable`, which I think can lead to inconsistent states, more specifically, `Get<RouterTable>().GetLeader() == NULL`. 

Since there is no `RouteTLV` in `Data Response`, here the fix creates a minimal `RouterIdSet` and updates the RouterTable. 

Another fix is call `BecomeDetached()` here? Thoughts? @librasungirl @jwhui

Reply:
Maybe go one step further - ignore MLE Data Response from different partition on FTD. Wait for MLE Advertisement as @librasungirl suggested.

Reply:
The solution sounds good to me. Will update according.

Reply:
Updated accordingly.